operator/src/main/java/oracle/kubernetes/operator/StuckPodProcessing.java

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodList;
+import oracle.kubernetes.operator.calls.CallResponse;
+import oracle.kubernetes.operator.helpers.CallBuilder;
+import oracle.kubernetes.operator.helpers.PodHelper;
+import oracle.kubernetes.operator.steps.DefaultResponseStep;
+import oracle.kubernetes.operator.work.NextAction;
+import oracle.kubernetes.operator.work.Packet;
+import oracle.kubernetes.operator.work.Step;
+import oracle.kubernetes.utils.SystemClock;
+import org.joda.time.DateTime;
+
+public class StuckPodProcessing {
+
+
+  private final MainDelegate mainDelegate;
+
+  public StuckPodProcessing(MainDelegate mainDelegate) {
+    this.mainDelegate = mainDelegate;
+  }
+
+  void checkStuckPods(String namespace) {
+    Step step = new CallBuilder()
+          .withLabelSelectors(LabelConstants.getCreatedbyOperatorSelector())
+          .listPodAsync(namespace, new PodListProcessing(namespace, SystemClock.now()));
+    mainDelegate.runSteps(step);
+  }
+
+  @SuppressWarnings("unchecked")
+  private List<V1Pod> getStuckPodList(Packet packet) {
+    return (List<V1Pod>) packet.computeIfAbsent("STUCK_PODS", k -> new ArrayList<>());
+  }
+
+  class PodListProcessing extends DefaultResponseStep<V1PodList> {
+
+    private final DateTime now;
+
+    public PodListProcessing(String namespace, DateTime dateTime) {
+      super(new PodActionsStep(namespace));
+      now = dateTime;
+    }
+
+    @Override
+    public NextAction onSuccess(Packet packet, CallResponse<V1PodList> callResponse) {
+      callResponse.getResult().getItems().stream()
+            .filter(pod -> isStuck(pod, now))
+            .forEach(pod -> addStuckPodToPacket(packet, pod));
+      
+      return doContinueListOrNext(callResponse, packet);
+    }
+
+    private boolean isStuck(V1Pod pod, DateTime now)  {
+      return getExpectedDeleteTime(pod).isBefore(now);
```

Comment:
- The stuck check potentially involves two different machine clocks that might be out of sync - the operator's and the pod's. I just did some brief research, and it looks like k8s doesn't require strict clock synchronization although it's considered desirable (in addition, WL clusters have fairly tight needs).
- Should there be an additional grace after a detecting a potentially stuck delete in order to give K8S time to 'naturally' honor its timeout and perhaps to help account for a 'small' amount of potential clock drift?

Reply:
That seems a reasonable precaution.

Reply:
Tough question. Off the top my head, maybe account for a few seconds of drift -- I've seen much worse on occasion, but, in general, WL clusters themselves dislike a drift any higher than that in my experience.   Plus perhaps account for a few extra seconds to give k8s time to handle the delete 'naturally'.  So that would total 10 seconds? And perhaps have an associated configurable?

@rjeberhard Thoughts?

Reply:
I keep any accounting for drift very small; perhaps just 1 or 2 seconds. It's subjective, but when I tested the solution it still "felt" like I waited a long time before the terminated pod was removed.

Reply:
Actually, what's the harm? If we delete a pod that is going to be deleted anyway, it should still be fine, right?

Reply:
It's my understanding this change initiates a forced delete - so in-memory-replicated-state data loss is possible if the forced delete occurs too early  for the operator to perform it's graceful shutdown

Reply:
The harm is that instead of catching only a pod that is "stuck" in the Terminating state that we could also detect pods that are in the process of shutting down normally and delete them before they've had a chance to complete normal shut down. This should only happen if the operator's pod is badly skewed from the master and the operator thinks the time is later than the master does.

Reply:
I wish that we could find a way to know that time at the master, but I've not been able to find it.